### PR TITLE
[bt#5455] Mails per Company und Signaturen

### PR DESCRIPTION
**Pull request by BT-tpiric**
bt#5455 mails per company and signatures - first commit

 
<div> Links to Odoo: </div> <ul>
<li><a target="_blank" href="https://odoo.braintec-group.com/web#view_type=form&model=helpdesk.ticket&id=5455">[bt#5455] Mails per Company und Signaturen</a></li>
</ul>

### CODE REVIEW

File: addons/mail/models/mail_template.py [hunk 1]
```diff
@@ -372,6 +372,33 @@ def render_template(self, template_txt, model, res_ids, post_process=False):
...
+                object_user_signature = record.user_id and \
+                    record.user_id.signature or False
+            if 'create_uid' in self.env[model]._fields:
+                object_create_user_signature = record.create_uid and \
+                    record.create_uid.signature or False
+            current_user_signature = self.env.user.signature or False
+            if 'company_id' in self.env[model]._fields and record.company_id:
+                res_users_obj_sudo = self.env['res.users'].sudo().with_context(
+                    force_company=record.company_id.id)
+                if record.user_id:
```
Comment (BT-sschmid) [inline addons/mail/models/mail_template.py]: maybe we should test if the fields exist too?

File: addons/mail/models/mail_template.py [hunk 2]
```diff
@@ -372,6 +372,33 @@ def render_template(self, template_txt, model, res_ids, post_process=False):
...
+                    object_user_signature = res_users_obj_sudo.browse(
+                        record.user_id.id).signature
+                if record.create_uid:
```
Comment (BT-sschmid) [inline addons/mail/models/mail_template.py]: here too?

File: addons/mail/wizard/mail_compose_message.py [hunk 3]
```diff
@@ -355,7 +355,16 @@ def onchange_template_id(self, template_id, composition_mode, model, res_id):
             if template.mail_server_id:
                 values['mail_server_id'] = template.mail_server_id.id
             if template.user_signature and 'body_html' in values:
-                signature = self.env.user.signature
+                if self.env[model].browse(res_id).company_id:
```
Comment (BT-sschmid) [inline addons/mail/wizard/mail_compose_message.py]: what if company id does not exist?

File: addons/mail/models/mail_template.py (hunk 1 above)
Comment (BT-tpiric) [reply addons/mail/models/mail_template.py]: resolved, thanks

File: addons/mail/models/mail_template.py (hunk 2 above)
Comment (BT-tpiric) [reply addons/mail/models/mail_template.py]: resolved, thanks

File: addons/mail/wizard/mail_compose_message.py (hunk 3 above)
Comment (BT-tpiric) [reply addons/mail/wizard/mail_compose_message.py]: resolved, thanks

**Pull request merged**